server/src/printers/TestMakefilesPrinter.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2012-2021. All rights reserved.
+ */
+
+#include "TestMakefilesPrinter.h"
+#include "utils/FileSystemUtils.h"
+
+#include <utility>
+#include <utils/MakefileUtils.h>
+
+namespace printer {
+    fs::path get_shared(fs::path path) {
+        return Paths::addSuffix(Paths::addPrefix(path, "__"), "_shared");
+    }
+
+    fs::path get_obj(fs::path path) {
+        return Paths::addSuffix(Paths::addPrefix(path, "__"), "_obj");
+    }
+
+    void Makefiles::write() const {
+        FileSystemUtils::writeToFile(path, generalMakefileStr);
+        fs::path sharedMakefilePath = get_shared(path);
+        FileSystemUtils::writeToFile(sharedMakefilePath, sharedMakefileStr);
```

Comment:
Why both `string` and `file` of the same entity are needed?

Reply:
path to file and content to write